internal/dcache/file_manager/file_manager.go

```diff
@@ -720,11 +723,14 @@ func (file *DcacheFile) readChunkWithReadAhead(offset int64) (*StagedChunk, erro
 	// We do it only when reading the start of a chunk.
 	//
 	if isOffsetChunkStarting(offset, &file.FileMetadata.FileLayout) {
-		for i := chunkIdx + 1; i <= readAheadEndChunkIdx; i++ {
+		// Start readahead after the last chunk readahead by prev read calls.
+		readAheadStartChunkIdx := max(file.lastReadaheadChunkIdx+1, chunkIdx+1)
+		for i := readAheadStartChunkIdx; i <= readAheadEndChunkIdx; i++ {
 			_, err := file.readChunk(i*file.FileMetadata.FileLayout.ChunkSize, false /* sync */)
 			if err != nil {
 				return nil, err
 			}
+			file.lastReadaheadChunkIdx = i
```

Comment:
If this can be called from multiple contexts, need to make lastReadaheadChunkIdx atomic